pkg/http/api.go

```diff
@@ -203,7 +203,7 @@ func (a *api) constructSecretEndpoints() []Endpoint {
 		},
 		{
 			Methods: []string{fasthttp.MethodGet},
-			Route:   "secrets/{secretStoreName}/{key}",
```

Comment:
Does this error happen even when "/" is encoded to "%2F"?

Reply:
When "/" is encoded to "%2F", it works fine.  I just notice the pubsub publish topic is handle like that, and when "/" is not encoded it doesn't work, so i think it would be treated as the same. Since it can work, that is not necessary.

Reply:
There is another question I don't know if it is deliberately designed like this.Also in valut secret component.

In GetSecret `vaultSecretPathAddr := fmt.Sprintf("%s/v1/secret/data/%s/%s?version=0", v.vaultAddress, v.vaultKVPrefix, secret)` while in BulkGetSecret `vaultSecretsPathAddr := fmt.Sprintf("%s/v1/secret/metadata/%s", v.vaultAddress, v.vaultKVPrefix)`.

This lead to a strange situation: when BulkGetSecret is called, the keys listed are under secret/metadata/{vaultKVPrefix}, but BulkGetSecret called GetSecret to find the value of key, which find the key secret/data/{vaultKVPrefix}. Two keys are different.

Reply:
I'll open another issue in component-contib to discuss [this problem](https://github.com/dapr/dapr/pull/3091#discussion_r619551951) and this pr can be closed now. If there is no other discussion and is not closed I'll close it tomorrow.

Reply:
We expect special chars like "/" to be URL encoded to work properly. So, yes, this PR should not be merged. Thanks for bringing this up. Please, create a separate issue for the KV discussion - I cannot fully understand your question here and we should not hijack this PR with another discussion :)